flink-core/src/main/java/org/apache/flink/configuration/ConfigOption.java

```diff
@@ -195,6 +196,27 @@ public T defaultValue() {
 		return defaultValue;
 	}
 
+	/**
+	 * Checks whether this option has deprecated keys.
+	 * @return True if the option has deprecated keys, false if not.
+	 * @deprecated Replaced by {@link #hasFallbackKeys()}
+	 */
+	@Deprecated
+	public boolean hasDeprecatedKeys() {
+		return fallbackKeys != EMPTY;
```

Comment:
What if we just delegate it to `hasFallbackKeys`? Concretely

```java
/**
  * @deprecated Replaced by {@link #hasFallbackKeys()}
  */
@Deprecated
public boolean hasDeprecatedKeys() {
  return hasFallbackKeys();
}
```

I think it would not cause compile error when you bump the version and it is better that we keep code deprecated like deprecated code.

Reply:
Agree reuse `hasFallbackKeys` and I would like to remain the java doc of `hasDeprecatedKeys `, because it is used to describe the method of `hasDeprecatedKeys `, not `FallbackKey`.